### PR TITLE
limit cni in kube-system

### LINKED ISSUE #19791: Install istio-cni in namespace istio-cni failed
**Bug description**

Follow this document - https://github.com/istio/installer#install-istio-cni cannot install istio cni. it has such error:
```
Error creating: pods "istio-cni-node-" is forbidden: pods with system-cluster-critical priorityClass is not permitted in istio-cni namespace
```

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

### PR DESCRIPTION
**Pull request by clyang82**
Fixed https://github.com/istio/istio/issues/19791

it was resolved in 1.17 - Critical pods should not be limited to kube-system namespace
 https://github.com/kubernetes/kubernetes/issues/60596

Signed-off-by: Chun Lin Yang <clyang@cn.ibm.com>

### CODE REVIEW

**Comment by clyang82** [comment]
we have update the document in this PR - https://github.com/istio/istio.io/pull/4518

**Review by howardjohn — COMMENTED** [review]
I don't think this is right, shouldn't we just make release.namespace==kube-system? otherwise it can't be customized?

**Comment by howardjohn** [comment]
I think it will actually make the documentation you linked no longer
correct as well

**Comment by rlenglet** [comment]
LGTM. istio-cni must be deployed in kube-system due to the daemonset's priority class. K8s doesn't give any flexibility for high priority classes. The Helm install instructions did the right thing. Maybe that setup got lost in the new installer.

**Comment by howardjohn** [comment]
Why not keep it customizable but set the default to kube system? I think
that's possible. Then the default is correct, but it's consistent with
other components and allows people on latest kubernetes to not install in
kube system

**Comment by rlenglet** [comment]
This is not customizable. K8s requires pods with those priority classes to
run in kube-system. It won't start if this is started in any other
namespaces.

All other CNI plugins deployed as daemonsets run in kube-system for the
same reason. This is consistent.

**Comment by howardjohn** [comment]
according to the link that's no longer true in 1.17

**Comment by rlenglet** [comment]
Oh sorry got it. I'm good with keeping it configurable but default to
kube-system.

**Comment by clyang82** [comment]
the reason is to hardcode `kube-system` without supporting customizable is Istio 1.4 has been tested with Kubernetes releases 1.13, 1.14, 1.15. It is error-prone if we allows configurable. We can enable configurable when we support Kubernetes 1.17.

**Comment by howardjohn** [comment]
This in master branch which is 1.5 and we intend to support 1.17 as far as
I know

**Comment by howardjohn** [comment]
Note that all of our CI runs 1.17 as well

**Comment by clyang82** [comment]
Good to know that. Let me keep it configurable but default to
kube-system

File: istio-cni/templates/configmap-cni.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ kind: ConfigMap
 apiVersion: v1
 metadata:
   name: istio-cni-config
-  namespace: kube-system
+  namespace: {{ .Release.Namespace }}
```
Comment (rlenglet) [inline istio-cni/templates/configmap-cni.yaml]: I think it needs its own configuration variable.
When deploying on k8s ≤1.17, this namespace needs to be different from the `Release.Namespace` variable.

**Comment by howardjohn** [comment]
namespace is set per component in this model, it's not the same monolothic
helm chart we had before

**Comment by clyang82** [comment]
Yes. namespace is per-component - `iop kube-system istio-cni $IBASE/istio-cni/ ${ISTIO_CNI_ARGS}`

**Comment by cagataygurturk** [comment]
Hey, at the moment it is impossible to deploy CNI with istioctl because the CNI will be installed to istio-system. I have hopes that this PR would fix it.

**Review by richardwxn — COMMENTED** [review]
shouldn't we need to update the templates ns as well?

**Comment by howardjohn** [comment]
I think we want to modify data/profiles/default.yaml in operator instead

**Comment by clyang82** [comment]
I am working on that right now.

**Review by morvencao — APPROVED** [review]
The minimal doc change is OK according to the discussion.

**Pull request merged**